src/pages/Minion.jsx

```diff
@@ -196,8 +196,73 @@ const MinionDetails = ({ overview, currentDaoTokens }) => {
     });
   };
 
+  const sendToken = async (values, token) => {
+    setLoading(true);
+    const tokenService = TokenService({
+      tokenAddress: token.contractAddress,
+      chainID: daochain,
+    });
+
+    const amountWithDecimal = +values.amount * 10 ** +token.decimals;
+
+    const hexData = tokenService('transferNoop')({
+      to: values.destination,
+      amount: amountWithDecimal.toString(),
+    });
+
+    const details = detailsToJSON({
+      title: `${minionData.details} sends a token`,
+      description: `Send ${values.amount} ${token.symbol}`,
+      // link: (link to block explorer)
+      type: 'tokenSend',
+    });
+    const args = [
+      token.contractAddress,
+      '0',
+      hexData,
+      details,
+    ];
+    try {
+      const poll = createPoll({ action: 'minionProposeAction', cachePoll })({
+        minionAddress: minionData.minionAddress,
+        createdAt: now,
+        chainID: daochain,
+        actions: {
+          onError: (error, txHash) => {
+            errorToast({
+              title: 'There was an error.',
+            });
+            resolvePoll(txHash);
+            console.error(`Could not find a matching proposal: ${error}`);
+          },
+          onSuccess: (txHash) => {
+            successToast({
+              title: 'Minion proposal submitted.',
+            });
+            refreshDao();
+            resolvePoll(txHash);
+          },
+        },
+      });
+      const onTxHash = () => {
+        setGenericModal(false);
+        setTxInfoModal(true);
+      };
+      await MinionService({
+        web3: injectedProvider,
+        minion,
+        chainID: daochain,
+      })('proposeAction')({
+        args, address, poll, onTxHash,
+      });
+    } catch (err) {
+      setLoading(false);
+      console.log('error: ', err);
+    }
+  };
+
   const action = {
-    send: (args) => setGenericModal({ Erc20end: true, args }),
+    sendToken,
```

Comment:
I'm curious why we need to wrap this function in an object. Since objects are mutable, it's better to pass the functions down directly than to store as methods. There could be something I'm missing though.